sklearn/metrics/classification.py

```diff
@@ -454,7 +461,9 @@ def jaccard_similarity_score(y_true, y_pred, normalize=True,
             score = pred_and_true / pred_or_true
             score[pred_or_true == 0.0] = 1.0
     else:
-        score = y_true == y_pred
+        C = confusion_matrix(y_true, y_pred, sample_weight=sample_weight)
+        den = C.sum(0) + C.sum(1) - C.diagonal()
+        score = C.diagonal()/den
 
     return _weighted_sum(score, sample_weight, normalize)
```

Comment:
Can this still work??

Reply:
I don't seem to understand as to why this might be a problem.

Reply:
Well in the previous code, score was an array of length n_samples. Now score is an array of length n_classes. It clearly cannot be weighted by sample_weight which is an array of length n_samples.